src/transforms/filter.jl

```diff
@@ -69,22 +65,50 @@ DropMissing(cols::T...) where {T<:ColSelector} =
 
 isrevertible(::Type{<:DropMissing}) = true
 
-_ftrans(::DropMissing{Colon}, table) =
+# ftrans
+_ftrans(::DropMissing{Colon}, cols) =
   Filter(row -> all(!ismissing, row))
 
-function _ftrans(transform::DropMissing, table)
-  allcols = Tables.columnnames(table)
-  cols = _filter(transform.colspec, allcols)
+_ftrans(::DropMissing, cols) =
   Filter(row -> all(!ismissing, getindex.(Ref(row), cols)))
+
+# nonmissing 
+_nonmissing(::Type{T}, c) where {T} = c
+_nonmissing(::Type{Union{Missing,T}}, c) where {T} =
+  collect(T, c)
+
+function _nonmissing(columns, col)
+  c = Tables.getcolumn(columns, col)
+  _nonmissing(eltype(c), c)
 end
 
 function apply(transform::DropMissing, table)
-  ftrans = _ftrans(transform, table)
+  allcols = Tables.columnnames(table)
+  cols    = _filter(transform.colspec, allcols)
+  ftrans  = _ftrans(transform, cols)
   newtable, fcache = apply(ftrans, table)
-  newtable, (ftrans, fcache)
+
+  # post-processing
+  columns = Tables.columns(newtable)
+  newcols = [col ∈ cols ? _nonmissing(columns, col) : Tables.getcolumn(columns, col)
+             for col in allcols]
+  𝒯 = (; zip(allcols, newcols)...)
+  ptable = 𝒯 |> Tables.materializer(newtable)
+
+  types = Tables.schema(table).types
+  ptable, (ftrans, fcache, types)
 end
 
 function revert(::DropMissing, newtable, cache)
-  ftrans, fcache = cache
-  revert(ftrans, newtable, fcache)
+  ftrans, fcache, types = cache
+
+  # pre-processing
+  columns = Tables.columns(newtable)
+  allcols = Tables.columnnames(newtable)
+  newcols = [collect(T, Tables.getcolumn(columns, col))
+             for (T, col) in zip(types, allcols)]
```

Comment:
Try to place this into a single line or introduce new intermediate variables so that the line is short enough.

Reply:
I used the Tables.columntable function to simplify the code.